test/decimal128_test.dart

```diff
@@ -0,0 +1,305 @@
+////////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////////
+
+import 'dart:math';
+
+import 'package:meta/meta.dart';
+import '../lib/src/native/realm_core.dart';
+import 'package:test/test.dart';
+
+const int defaultTimes = 100000;
+
+void repeat(dynamic Function() body, [int times = defaultTimes]) {
+  for (var i = 0; i < times; ++i) {
+    body();
+  }
+}
+
+@isTest
+void repeatTest(String description, dynamic Function(Decimal128 x, int xInt, Decimal128 y, int yInt) body, [int times = defaultTimes]) {
+  final r = Random(42); // use a fixed seed to make tests deterministic
```

Comment:
Fixed, but for the record...

`r` is declared and used twice over just 7 lines.. this is **not** a readability issue. I will fix this just because, but please don't vilify all short variable names. They put emphasis on the _operations_ as opposed to the _variables_.

Or in the words of Bob Nyström (one of the Dart language designers): https://journal.stuffwithstuff.com/2016/06/16/long-names-are-long/